lib/core/sys/setup.cpp

```diff
@@ -0,0 +1,41 @@
+#include "core/sys/setup.h"
+
+#include <clang/Basic/Version.h>
+#include <spdlog/fmt/fmt.h>
+#include <spdlog/spdlog.h>
+
+#include <cstdlib>
+
+namespace {
+std::unique_ptr<std::string> CLANG_SYSROOT_DIR;
+std::unique_ptr<std::string> CLANG_ISYSTEM_OPT;
+
+constexpr char EB_ROOT_CLANG[] = "EBROOTCLANG";
+
+__attribute__((constructor)) void initClangEnvVars() {
+    CLANG_SYSROOT_DIR = std::make_unique<std::string>();
```

Comment:
Strings are already pointers to data (internally).
If you wish to make it properly, you may change `getSysRoot` and `getISystemOpt` return references to a string, that is statically allocated inside.

Reply:
Initially it was done in this manner, however it does not work.
Do not have time to dig into so used ptr instead.